wgpu-hal/src/gles/conv.rs

```diff
@@ -417,104 +417,74 @@ pub(super) fn map_storage_access(access: wgt::StorageTextureAccess) -> u32 {
     }
 }
 
-pub(super) fn is_sampler(glsl_uniform_type: u32) -> bool {
-    match glsl_uniform_type {
-        glow::INT_SAMPLER_1D
-        | glow::INT_SAMPLER_1D_ARRAY
-        | glow::INT_SAMPLER_2D
-        | glow::INT_SAMPLER_2D_ARRAY
-        | glow::INT_SAMPLER_2D_MULTISAMPLE
-        | glow::INT_SAMPLER_2D_MULTISAMPLE_ARRAY
-        | glow::INT_SAMPLER_2D_RECT
-        | glow::INT_SAMPLER_3D
-        | glow::INT_SAMPLER_CUBE
-        | glow::INT_SAMPLER_CUBE_MAP_ARRAY
-        | glow::UNSIGNED_INT_SAMPLER_1D
-        | glow::UNSIGNED_INT_SAMPLER_1D_ARRAY
-        | glow::UNSIGNED_INT_SAMPLER_2D
-        | glow::UNSIGNED_INT_SAMPLER_2D_ARRAY
-        | glow::UNSIGNED_INT_SAMPLER_2D_MULTISAMPLE
-        | glow::UNSIGNED_INT_SAMPLER_2D_MULTISAMPLE_ARRAY
-        | glow::UNSIGNED_INT_SAMPLER_2D_RECT
-        | glow::UNSIGNED_INT_SAMPLER_3D
-        | glow::UNSIGNED_INT_SAMPLER_CUBE
-        | glow::UNSIGNED_INT_SAMPLER_CUBE_MAP_ARRAY
-        | glow::SAMPLER_1D
-        | glow::SAMPLER_1D_SHADOW
-        | glow::SAMPLER_1D_ARRAY
-        | glow::SAMPLER_1D_ARRAY_SHADOW
-        | glow::SAMPLER_2D
-        | glow::SAMPLER_2D_SHADOW
-        | glow::SAMPLER_2D_ARRAY
-        | glow::SAMPLER_2D_ARRAY_SHADOW
-        | glow::SAMPLER_2D_MULTISAMPLE
-        | glow::SAMPLER_2D_MULTISAMPLE_ARRAY
-        | glow::SAMPLER_2D_RECT
-        | glow::SAMPLER_2D_RECT_SHADOW
-        | glow::SAMPLER_3D
-        | glow::SAMPLER_CUBE
-        | glow::SAMPLER_CUBE_MAP_ARRAY
-        | glow::SAMPLER_CUBE_MAP_ARRAY_SHADOW
-        | glow::SAMPLER_CUBE_SHADOW => true,
-        _ => false,
-    }
-}
-
-pub(super) fn is_image(glsl_uniform_type: u32) -> bool {
-    match glsl_uniform_type {
-        glow::INT_IMAGE_1D
-        | glow::INT_IMAGE_1D_ARRAY
-        | glow::INT_IMAGE_2D
-        | glow::INT_IMAGE_2D_ARRAY
-        | glow::INT_IMAGE_2D_MULTISAMPLE
-        | glow::INT_IMAGE_2D_MULTISAMPLE_ARRAY
-        | glow::INT_IMAGE_2D_RECT
-        | glow::INT_IMAGE_3D
-        | glow::INT_IMAGE_CUBE
-        | glow::INT_IMAGE_CUBE_MAP_ARRAY
-        | glow::UNSIGNED_INT_IMAGE_1D
-        | glow::UNSIGNED_INT_IMAGE_1D_ARRAY
-        | glow::UNSIGNED_INT_IMAGE_2D
-        | glow::UNSIGNED_INT_IMAGE_2D_ARRAY
-        | glow::UNSIGNED_INT_IMAGE_2D_MULTISAMPLE
-        | glow::UNSIGNED_INT_IMAGE_2D_MULTISAMPLE_ARRAY
-        | glow::UNSIGNED_INT_IMAGE_2D_RECT
-        | glow::UNSIGNED_INT_IMAGE_3D
-        | glow::UNSIGNED_INT_IMAGE_CUBE
-        | glow::UNSIGNED_INT_IMAGE_CUBE_MAP_ARRAY
-        | glow::IMAGE_1D
-        | glow::IMAGE_1D_ARRAY
-        | glow::IMAGE_2D
-        | glow::IMAGE_2D_ARRAY
-        | glow::IMAGE_2D_MULTISAMPLE
-        | glow::IMAGE_2D_MULTISAMPLE_ARRAY
-        | glow::IMAGE_2D_RECT
-        | glow::IMAGE_3D
-        | glow::IMAGE_CUBE
-        | glow::IMAGE_CUBE_MAP_ARRAY => true,
-        _ => false,
-    }
-}
+pub(super) fn map_naga_uniform_type(ty: &naga::TypeInner) -> u32 {
```

Comment:
I see this is only used for `uniform_byte_size` and the match statement.
Can we keep using `TypeInner` and its `size` method instead?

Reply:
Nice, didn't know that existed!